### PR TITLE
Replace most jQuery with vanilla JavaScript

### LINKED ISSUE #14596: forward port jquery removal from ipympl
https://github.com/matplotlib/jupyter-matplotlib/pull/85  removes jquery from ipympl.  Much of that js is shared / derived from the webagg and nbagg backends currently shipped in Matplotlib.

Going forward we should pull that patch forward (probably not for old version so it won't help us with #14587 and friends) and deprecate the nbagg backend (as it does not work with jlab) but keep the webagg backend.

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This is a mix of the change in ipympl, plus the same thing for our original version of other stuff, using stuff like http://youmightnotneedjquery.com/ and [this cheat sheet](https://tobiasahlin.com/blog/move-from-jquery-to-vanilla-javascript/). Unfortunately, we can't just remove jQuery, because we have jquery-ui for the CSS and resizable handles. CSS is easy, and we can replace the handles without too much code (since we don't need its full flexibility), but this was large enough already.

Fixes #14596. ~~Waiting on the JS style fix PR.~~

## PR Checklist

- [x] Has Pytest style unit tests
- [N/A] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [N/A] New features are documented, with examples if plot related
- [N/A] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [N/A] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/web_backend/js/mpl.js [hunk 1]
```diff
@@ -140,15 +150,17 @@ mpl.figure.prototype._init_canvas = function () {
...
+    ));
+    rubberband_canvas.setAttribute(
         'style',
         'position: absolute; left: 0; top: 0; z-index: 1;'
     );
 
     var pass_mouse_events = true;
 
-    canvas_div.resizable({
+    $(canvas_div).resizable({
```
Comment (tacaswell) [inline lib/matplotlib/backends/web_backend/js/mpl.js:163]: Is this putting jquery back in?

File: lib/matplotlib/backends/web_backend/js/mpl.js [hunk 2]
```diff
@@ -253,60 +282,48 @@ mpl.figure.prototype._init_toolbar = function () {
...
-        }).html(fmt);
-        fmt_picker.append(option);
+        var option = document.createElement('option');
+        option.selected = fmt === mpl.default_extension;
+        option.innerHTML = fmt;
+        fmt_picker.appendChild(option);
     }
 
-    // Add hover states to the ui-buttons
-    $('.ui-button').hover(
```
Comment (tacaswell) [inline lib/matplotlib/backends/web_backend/js/mpl.js:298]: It looks like this is gone?  Do we care or did it get moved someplace else?

File: lib/matplotlib/backends/web_backend/js/mpl.js (hunk 1 above)
Comment (QuLogic) [reply lib/matplotlib/backends/web_backend/js/mpl.js:163]: It's needed for the only usage of jQuery UI. That'll be removed in the last PR.

File: lib/matplotlib/backends/web_backend/js/mpl.js (hunk 2 above)
Comment (QuLogic) [reply lib/matplotlib/backends/web_backend/js/mpl.js:298]: It didn't seem to do anything; we can do hover with CSS now.

Comment (tacaswell) [reply lib/matplotlib/backends/web_backend/js/mpl.js:298]: it seems to still work so without this so I guess we are not worried.

File: lib/matplotlib/backends/web_backend/js/mpl.js (hunk 1 above)
Comment (tacaswell) [reply lib/matplotlib/backends/web_backend/js/mpl.js:163]: I see, this gets taken back out in https://github.com/matplotlib/matplotlib/pull/17086/commits/b398c0d6b416e6501c47e34cc301e1aefeaab148

**Comment by tacaswell** [comment]
I went over this with @dorafc who signed off on the JS.  Did local testing, something is broken with nabagg as closing the figure on the python side does not close the js side, but that is also broken on master.

File: lib/matplotlib/backends/web_backend/js/mpl.js (hunk 2 above)
Comment (tacaswell) [reply lib/matplotlib/backends/web_backend/js/mpl.js:298]: sorry, I had not refreshed an did not see your comment before I responded again :sheep:

**Comment by tacaswell** [comment]
The circle failure is due to the dateutil inventory not downloading and travis is due to a weird wx import error.

**Comment by QuLogic** [comment]
Circle might be helped a bit by caching in #17183.

**Pull request merged**